docs/reviewer-guide.md

```diff
@@ -41,6 +41,58 @@ Make a few passes over the code you want to review.
 * List ordering on CRD APIs matter. If two components need to update a list on the same object, make sure both components do it in a way that preserves the order of the list. For example, both virt-handler and virt-controller need to modify conditions on the VMI status. If both virt-handler and virt-controller are constantly changing the order of the conditions list, that will cause an update storm where both components are competing with one another to write changes.
 * Privileged node-level operations should be added to virt-handler and not virt-launcher to keep the privileges on virt-launcher at a minimum.
 
+## In-Depth PATCH/UPDATE considerations
+
+When trying to determine if an UPDATE or a PATCH should be used, the following has to be considered:
+
+### Updating `spec` or `metadata` sections with a controller
+
+Did the controller in question CREATE the object it operates on?
+- Yes: `Update` can be used
+- No: `Patch` with a narrow focus on the fields of interest needs to be used
```

Comment:
I get what you're saying. this might get confusing though. The problem is update isn't limited to spec/metadata except for a couple of our resources which have the status subresource

maybe this can be simplified to something like, "always use patch unless you're working in the controller that controls the resource's status"

Reply:
> I get what you're saying. this might get confusing though. The problem is update isn't limited to spec/metadata except for a couple of our resources which have the status subresource

Even the VMI which does not have a /status subresource has a `UpdateStatus` field to make things consistent for developers.

Let's discuss this again for resources like VMPool (I think that is the only one) which may not have a `UpdateStatus` and see if we should add it.

> maybe this can be simplified to something like, "always use patch unless you're working in the controller that controls the resource's status"

The section here is about the `spec`. Like the VM Pool creating VM objects. I think I can't simplify it to that. The case you describe is covered below in the section about updating the `status`.

Reply:
@davidvossel could you have a look at my answer above?